### PR TITLE
fix: iterate over each document definition once

### PR DESCRIPTION
**Pull request by dpilch**
#### Description of changes

Iterate over each document definition once. Previously documents were iterated over `numTotalDocuments - index` times.

This fixes a bug that arises when multiple schema files are used with multiple `extends`.

`schema/a.graphql`
```
type Query {
    queryA: String!
}
```

`schema/b.graphql`
```
extend type Query {
    queryB: String!
}
```

`schema/c.graphql`
```
extend type Query {
    queryC: String!
}
```

The definition in `schema/b.graphql` was processed twice resulting in error: `🛑 Object type extension 'Query' cannot redeclare field queryB`

##### CDK / CloudFormation Parameters Changed

N/A

#### Issue #, if available

Resolves https://github.com/aws-amplify/amplify-category-api/issues/1808

#### Description of how you validated changes

Unit testing

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-transformer-core/src/transformer-context/output.ts [hunk 1]
```diff
@@ -297,13 +344,17 @@ export class TransformerOutput implements TransformerContextOutputProvider {
...
+    this.ensureSubscription();
     const subscriptionTypeName = this.getSubscriptionTypeName();
     if (subscriptionTypeName) {
       if (!this.getType(subscriptionTypeName)) {
         this.addType(blankObject(subscriptionTypeName));
       }
       let subscriptionType = objectExtension(subscriptionTypeName, fields);
       this.addObjectExtension(subscriptionType);
+    } else {
+      this.addDefaultSubscription();
```
Comment (alharris-at) [inline packages/amplify-graphql-transformer-core/src/transformer-context/output.ts:356]: This should be covered by the call to `this.ensureSubscription()`, right?

Comment (dpilch) [reply packages/amplify-graphql-transformer-core/src/transformer-context/output.ts:356]: Oops, yes. This is the original implementation before I changed it use `ensureSubscription`.

**Review by AaronZyLee — APPROVED** [review]
Overall LGTM. Could you also double check by `amplify-dev` with the multiple schema files using `extend`? (same as the case in the issue)

**Review by alharris-at — APPROVED** [review]

**Comment by dpilch** [comment]
Holding off on merging until construct release.

**Pull request merged**